### ISSUE TITLE
Python token bucket rate limit algorithm for non async client

### ISSUE THREAD

**Issue by hjastenger** [ccxt/ccxt#9374]
- OS: NIX
- Programming Language version: Python3.9
- CCXT version: Latest

Hi,

I have been going over all the comments and issues related to how rate limiting has currently been setup for ccxt. This is by no means a way to bash at you guy, just wondering if some of my assumptions are correct after comparing the code base and the documentation.

Looking at all the information from the rate limit issue (https://github.com/ccxt/ccxt/issues/1399#issuecomment-359233015) and the current code base I get the impression that the current SYNC python client only supports the 'simpler' and naiver rate limiting logic, while the ASYNC client has support for the more advanced 'token bucket' implementation.

Long story short: if you want to use the 'token bucket' algorithm for rate limiting in Python you have to use the ASYNC client.

Could you maybe confirm this?

**Comment by ttodua** [comment]
seems an old topic, left without comment, as OP was correct in what he described.
closing this.
